homework/shared_ptr/test.cpp

```diff
@@ -97,7 +97,56 @@ TEST_F(weakPtrTest, shouldCopyAssign) {
 }
 
 TEST_F(weakPtrTest, testLock) {
-    cs::weak_ptr weak(sPtr);
+    cs::weak_ptr<int> wPtr(sPtr);
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+}
+
+TEST_F(weakPtrTest, testCopyConstructorFromShared) {
+    cs::weak_ptr<int> wPtr(sPtr);
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+    ASSERT_EQ(wPtr.use_count(), 1);
+}
+
+TEST_F(weakPtrTest, testCopyConstructorFromWeak) {
+    cs::weak_ptr<int> wPtr(sPtr);
+    auto wPtr2(wPtr);
+    ASSERT_FALSE(wPtr.expired());
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+    ASSERT_EQ(wPtr.use_count(), 2);
+}
+
+TEST_F(weakPtrTest, testCopyAssignment) {
+    cs::weak_ptr<int> wPtr = sPtr;
+    ASSERT_EQ(*(wPtr.lock()), testValueOne);
+    ASSERT_EQ(wPtr.use_count(), 1);
+}
+
+TEST_F(weakPtrTest, testMoveConstructorFromShared) {
```

Comment:
You do not move from shared to weak, like ever. And this code is the same as copy c-tor

Reply:
rozpędziłęm się :)
test removed